GenderPayGap.WebUI/Classes/Extensions/HtmlHelpers.cs

```diff
@@ -62,9 +63,11 @@ public static async Task<IHtmlContent> PartialModelAsync<T>(this IHtmlHelper htm
         public static string WithQuery(this IUrlHelper helper, string actionName, object routeValues)
         {
             var newRoute = new NameValueCollection();
-            foreach (string key in helper.ActionContext.HttpContext.Request.Query.Keys)
+
+            if (helper.ActionContext.HttpContext.Request.QueryString.HasValue)
             {
-                newRoute[key] = helper.ActionContext.HttpContext.Request.Query[key];
+                var existingRoute = HttpUtility.ParseQueryString(helper.ActionContext.HttpContext.Request.QueryString.Value);
+                newRoute.Add(existingRoute);

```

Comment:
Nice! Have you tested it with any other more complex query strings elsewhere?

Reply:
I tested it locally, on the search results page. Any other ideas?

Reply:
Search results is the most complicated GET query I can think of in the site, so this works for me :+1: